include/ck_tile/ops/fmha/pipeline/block_fmha_bwd_pipeline_default_policy.hpp

```diff
@@ -52,21 +59,27 @@ struct BlockFmhaBwdPipelineDefaultPolicy
                                                 typename Problem::BlockFmhaShape::Gemm0BlockWarps,
                                                 WarpGemm>;
 
-        return BlockGemmARegBRegCRegV1<BlockGemmProblem, BlockGemmPolicy>{};
+        return BlockGemmARegBRegCRegV1<GemmProblem, BlockGemmPolicy>{};
     }
 
     template <typename Problem>
     CK_TILE_HOST_DEVICE static constexpr auto GetPTOGradTBlockGemm()
     {
-        using BlockGemmProblem = BlockGemmPipelineProblem<
-            typename Problem::GemmDataType,
-            typename Problem::OGradDataType,
-            typename Problem::AccDataType,
-            TileGemmShape<sequence<Problem::BlockFmhaShape::kN0,
-                                   Problem::BlockFmhaShape::kVHeaddim,
-                                   Problem::BlockFmhaShape::kK1>,
-                          typename Problem::BlockFmhaShape::Gemm1BlockWarps,
-                          typename Problem::BlockFmhaShape::Gemm1WarpTile>>;
+        using GemmProblem =
+            GemmPipelineProblem<typename Problem::GemmDataType,
+                                typename Problem::OGradDataType,
+                                typename Problem::AccDataType,
+                                TileGemmShape<sequence<Problem::BlockFmhaShape::kN0,
+                                                       Problem::BlockFmhaShape::kVHeaddim,
+                                                       Problem::BlockFmhaShape::kK1>,
+                                              typename Problem::BlockFmhaShape::Gemm1BlockWarps,
+                                              typename Problem::BlockFmhaShape::Gemm1WarpTile>,
+                                TileGemmTraits<Problem::kPadSeqLenQ,
```

Comment:
These TileGemmTraits are useless in fmha fwd&bwd policy, and the kPadX_ & LayoutX_ may not match the real Gemm implementation in fmha and can be confusing. Maybe you can set kPadX_ to false and LayoutX_ to RowMajor with comment that these settings are useless. @qianfengz @poyenc What do you think?

Reply:
Thank you for the comment Dan! I just discussed with Qianfeng offline. I will call the Block level BlockGemmProblem to avoid the useless information. The fmha only cares about block level gemm so I will let it call the block-level gemm  problem. All the previous version call it on the pipeline level. Thank you for the comment!